src/misc/lv_event.c

```diff
@@ -153,19 +172,19 @@ bool lv_event_remove(lv_event_list_t * list, uint32_t index)
 {
     LV_ASSERT_NULL(list);
     lv_event_dsc_t * dsc = lv_event_get_dsc(list, index);
-    lv_free(dsc);
-    return lv_array_remove(list, index);
+    if(dsc == NULL) return false;
+    event_mark_deleting(list, dsc);
+    cleanup_event_list(list);
+    return true;
```

Comment:
Original logic will check if `index` is really in list, and return false if not.
```c
return lv_array_remove(list, index); // here will check if index is valid.
```

Reply:
previous line: 
```cpp
lv_event_dsc_t * dsc = lv_event_get_dsc(list, index);
if(dsc == NULL) return false;
```
will check this